### ISSUE TITLE
Why is UINib.instantiateWithOwner(_:options:) needed in viewDidLoad()?

### ISSUE THREAD

**Issue by JohnEstropia** [jessesquires/JSQMessagesViewController#1703]
## Question
Why does `JSQMessagesViewController` need to load it's nib from within `viewDidLoad`?
https://github.com/jessesquires/JSQMessagesViewController/blob/develop/JSQMessagesViewController/Controllers/JSQMessagesViewController.m#L239

Wouldn't this mess up subclasses' initialization?

**Comment by jessesquires** [comment]
Hey @JohnEstropia ! 👍 

No, subclasses can still customize the view in the usual ways. Subclasses can also override the `nib` so that their custom nib is loaded.

**Comment by JohnEstropia** [comment]
This was closed a few days ago but I just want to point out that this
```
- (void)viewDidLoad
{
    [super viewDidLoad];

    [[[self class] nib] instantiateWithOwner:self options:nil];

```
causes an errant JSQMessagesViewController instance (owned by the UINib) to be temporarily created and immediately deallocated after viewDidLoad. Normally this would be fine, but we've hit a few roadblocks here because in our case the temporary instance caused imbalanced KVO registration and unregistration. 

We've worked around it somehow but some users may find this explanation useful.
